fcrepo/fcrepo
# FCREPO-3950 - Prevent containment of fcr:acl on reindex

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3950

# What does this Pull Request do?

Adds tests to replicate the bug, and then prevents adding containment relations (and by extension, membership relations) for fcr:acl resources. This applies to basic, direct and indirect containers.

# How should this be tested?

See the ticket for instructions on replicating the issue. Effectively, create a container, add an fcr:acl to it, and then rebuild the index. The new tests should replicate the issue too.

# Interested parties
@fcrepo/committers
